crypto/pem/pem_lib.c

```diff
@@ -537,12 +536,13 @@ int PEM_get_EVP_CIPHER_INFO(char *header, EVP_CIPHER_INFO *cipher)
     *header = '\0';
     cipher->cipher = enc = EVP_get_cipherbyname(dekinfostart);
     *header = c;
+    header++;
```

Comment:
And the original code “looked wrong” which is what started you down this path.  Star-p-plus-plus is the C idiom.  Not two statements.